lib/PermissionedRoute/PermissionedRoute.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { Route } from 'react-router-dom';
+import PropTypes from 'prop-types';
+
+import { IfPermission } from '@folio/stripes/core';
+
+import NoPermissionsMessage from './NoPermissionsMessage';
+
+const PermissionedRoute = ({ children, perm, returnLinkLabelId, returnLink, ...rest }) => (
+  <Route
+    {...rest}
+    render={() => (
+      <IfPermission perm={perm}>
+        {({ hasPermission }) => (hasPermission
+          ? children
+          : (
+            <NoPermissionsMessage
+              returnLinkLabelId={returnLinkLabelId}
+              returnLink={returnLink}
+            />
+          )
+        )}
+      </IfPermission>
```

Comment:
```suggestion
    stripes.hasPerm(perm)
        ? children
        : <NoPermissionsMessage />
```

Reply:
Actually, it's the same. In your suggestion `PermissionedRoute` component should be connected to `stripes`

Reply:
```
import { useStripes } from '@folio/stripes/core';
```

then, in the component,
```
const stripes = useStripes();
```
I think, it should work. Looks cleaner too.